src/renderer/store/app/index.ts

```diff
@@ -70,7 +68,6 @@ export const useApp = create<StoreState & StoreReducers>((set, get) => ({
     }
 
     await Promise.all(promises);
-    set({ initialized: true });
```

Comment:
This seems like a problem if we only complete initialization on `downloadDolphinFinished`. What happens if you disconnect from the internet and start the app? The download would never complete and initialized would never be set.

Reply:
I think one solution is to have the `downloadDolphinFinished` receive an optional error object. Then send that error object along with the `downloadDolphinFinished` event if an error occurs.